### PR TITLE
Support testing in toolbox

### PR DESCRIPTION
**Pull request by mfocko**
- Add Makefile target for
  - Setting up from inside toolbox
  - Testing in toolbox (checks if exists + sets it up if necessary)
- Script that checks if toolbox exists, otherwise sets it up
- Playbook with necesary targets

Signed-off-by: Matej Focko <mfocko@redhat.com>

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -90,5 +90,12 @@ check-inside-openshift: service worker test_image
 check-inside-openshift-zuul: test_image
 	ANSIBLE_STDOUT_CALLBACK=debug $(AP) files/check-inside-openshift.yaml
 
+setup-toolbox:
+	dnf install -y ansible
```
Comment (lachmanfrantisek) [inline Makefile:94]: I am not sure if it is necessary. We don't do that elsewhere.

**Review by lachmanfrantisek — COMMENTED** [review]
Nice!

File: Makefile (hunk 1 above)
Comment (mfocko) [reply Makefile:94]: fedora-toolbox image doesn't have ansible preinstalled

File: files/test_in_toolbox.sh [hunk 2]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+
```
Comment (csomh) [inline files/test_in_toolbox.sh:2]: `set -eu` to fail on the first error and when using undefined variables.

**Review by csomh — COMMENTED** [review]
I'm not sure I understand what the intended utility for these targets is... Why not just use one of the existing images to create a toolbox? How is `check-inside-toolbox` different from `check_in_container`? (besides using `toolbox` vs `podman`/`docker`)

**Comment by mfocko** [comment]
> I'm not sure I understand what the intended utility for these targets is... Why not just use one of the existing images to create a toolbox? How is `check-inside-toolbox` different from `check_in_container`? (besides using `toolbox` vs `podman`/`docker`)

@csomh It seems to me it's more swift compared to running in container where I've encountered rebuilds of image even throughout the day
(also it was more straightforward to set up local mirror for toolbox rather than `usercont/base` that managed to keep both images from registry and local and continued to use local ones -__-)

**Comment by lachmanfrantisek** [comment]
> Why not just use one of the existing images to create a toolbox?

I don't think this will be easy. The toolbox images do not have easy build instructions.

> How is check-inside-toolbox different from check_in_container?

For users that don't use toolbox, there is probably only a small benefit (maybe easier access to the test environment).

Otherwise, it's more a question of "Why use toolbox at all?." For me, it's really handy, especially because I am using Silveblue and have all the development tools (GUI included) inside toolbox.

File: Makefile (hunk 1 above)
Comment (lachmanfrantisek) [reply Makefile:94]: Aha, this is not run on the host? Then I would suggest some better name.

**Comment by csomh** [comment]
> I don't think this will be easy. The toolbox images do not have easy build instructions.

Strange, I thought something like `toolbox create --container packit-service --image usercont/packit-service:prod` is enough. Trying it out seems to be working and it seems to me that it's the right environment. What am I missing?

**Comment by mfocko** [comment]
> > I don't think this will be easy. The toolbox images do not have easy build instructions.
> 
> Strange, I thought something like `toolbox create --container packit-service --image usercont/packit-service:prod` is enough. Trying it out seems to be working and it seems to me that it's the right environment. What am I missing?

pytest and flexmock from `files/recipe-tests.yaml` (you probably have them installed in `$HOME` through pip? iirc there were some problems with pytest from pip instead of RPM, maybe they're resolved)

File: Makefile (hunk 1 above)
Comment (mfocko) [reply Makefile:94]: what name do you suggest? :)

I could also move ansible installation into the playbook and add check for `/run/.envrc` to make sure we're running in container / toolbox

Comment (lachmanfrantisek) [reply Makefile:94]: `setup-inside-toolbox`/`install-dependecies-into-toolbox`/`dependencies-inside-toolbox`/..

I now understand how your name makes sense but was confused originally if we are setting-up the toolbox or making setup inside toolbox.

**Review by jpopelka — APPROVED** [review]

**Pull request merged**